packages/backend/src/services/GitIntegrationService/GitIntegrationService.ts

```diff
@@ -466,4 +481,131 @@ Affected charts:
+
+    private static async createSqlFile({
+        githubProps,
+        name,
+        sql,
+    }: {
+        githubProps: GithubProps;
+        name: string;
+        sql: string;
+    }) {
+        const fileName = `models/${snakeCase(name)}.sql`;
+        const content = `
+{{
+  config(
+    tags=['created-by-lightdash']
+  )
+}}
+  
+${sql}
+`;
+
+        return createFile({
+            ...githubProps,
+            fileName,
+            content,
+            message: `Created file ${fileName} `,
+        });
+    }
+
+    private static async createYmlFile({
+        githubProps,
+        name,
+        columns,
+    }: {
+        githubProps: GithubProps;
+        name: string;
+        columns: VizColumn[];
+    }) {
+        const fileName = `models/${snakeCase(name)}.yml`;
+        const content = yaml.dump(
+            {
+                version: 2,
+                models: [
+                    {
+                        name: snakeCase(name),
+                        label: friendlyName(name),
+                        description: `SQL model for friendlyName(${name})`,
+                        columns: columns.map((c) => ({
+                            name: c.reference,
+                            meta: {
+                                dimension: {
+                                    type: c.type,
+                                },
+                            },
+                        })),
+                    },
+                ],
+            },
+            {
+                quotingType: githubProps.quoteChar,
+            },
+        );
+
+        return createFile({
+            ...githubProps,
+            fileName,
+            content,
+            message: `Created file ${fileName} `,
+        });
+    }
+
+    async createPullRequestFromSql(
+        user: SessionUser,
+        projectUuid: string,
+        name: string,
+        sql: string,
+        columns: VizColumn[],
+        quoteChar: `"` | `'` = '"',
+    ): Promise<PullRequestCreated> {
+        const { owner, repo, branch, path } = await this.getProjectRepo(
+            projectUuid,
+        );
+
+        const token = await this.getOrUpdateToken(user.organizationUuid!);
+
+        const branchName = await GitIntegrationService.createBranch({
+            owner,
+            repo,
+            mainBranch: branch,
+            token,
+        });
+        const githubProps: GithubProps = {
+            owner,
+            repo,
+            branch: branchName,
+            token,
+            path,
+            quoteChar,
+        };
+        await GitIntegrationService.createSqlFile({
+            githubProps,
+            name,
+            sql,
+        });
+        await GitIntegrationService.createYmlFile({
+            githubProps,
+            name,
+            columns,
+        });
+        const pullRequest = await createPullRequest({
+            owner,
+            repo,
+            title: `Write back \`${name}\` SQL and YML model`,
```

Comment:
```suggestion
            title: `Write back \`${snakeCase(name)}\` SQL and YML model`,
```

can be done later/it's not that important